packages/web-app-files/tests/unit/views/spaces/DriveResolver.spec.ts

```diff
@@ -28,6 +33,24 @@ describe('DriveResolver view', () => {
     const { wrapper } = getMountedWrapper({ space: spaces[0] })
     expect(wrapper.find('generic-space-stub').exists()).toBeTruthy()
   })
+  it('redirects to the public drop page in a public context with "upload-only"-permissions', async () => {
+    const space = { id: '1', getDriveAliasAndItem: jest.fn(), driveType: 'public' }
+    const clientService = mockDeep<ClientService>()
+    clientService.webdav.getFileInfo.mockResolvedValue(
+      mockDeep<PublicSpaceResource>({ publicLinkPermission: SharePermissionBit.Create })
+    )
+    const { wrapper, mocks } = getMountedWrapper({
+      space,
+      mocks: { $clientService: clientService }
+    })
+
+    await wrapper.vm.$nextTick()
+    await wrapper.vm.$nextTick()
```

Comment:
I'm afraid we need this? but it's ugly :(

Reply:
Yes, I didn't find a way around it unfortunately :/